xfce.xfce4-power-manager: use fixed path for helpers

pjstadig opened the pull request:
Using a nix store path for `xfpm-power-backlight-helper` and `xfce4-pm-helper` causes permission errors after a rebuild when adjusting brightness or putting the system to sleep, because the permitted executables are different from the ones `xfce4-power-manager` wants to execute.  Using a fixed path (/run/current-system/sw/bin) means the paths do not change after a rebuild, so there's no permission error.

A similar approach is taken with gnome's `gsd-backlight-helper`.

###### Description of changes

Before building, the `sbindir` variable references in the polkit configuration and in the `xfce4-power-manager` source are replaced with the hard coded path `/run/current-system/sw/bin`. The polkit and executable paths will always match, even if the system is rebuilt while `xfce4-power-manager` is running. This approach was stolen from gnome's handling of `gst-backlight-helper`.

One alternative I had considered, but didn't fully explore, is somehow restarting `xfce4-power-manager` after a rebuild (using an `activationScript`?). If it would work, it would be a fix only for the NixOS module, which may or may not be better.

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

@bobby285271 @romildo @muscaln

bobby285271 commented:
Thanks, I saw your screenshots in https://github.com/thoughtfull-systems/nixfiles/issues/45 (which is linked in #77485) so I believe the issue is there, though it might take me a while to actually review and merge :bow:

> A similar approach is taken with gnome's `gsd-backlight-helper`.

I have the impression that the GSD's patch is later dropped in https://github.com/NixOS/nixpkgs/commit/c7cf166533f26a0aca9ce2a68c326af65b1cbb5d with the introduction of https://github.com/GNOME/gnome-settings-daemon/commit/cd051e6c7ccfb7e7b0ec440409c559b2e51284e8, probably this is not done in Xfce :thinking: 

---

And I actually have 2 small nits, do you mind adding a short comment (a link to https://github.com/NixOS/nixpkgs/issues/77485 or #245447 should be enough) to document the substitution & prefix the commit message with `xfce.xfce4-power-manager` so ofborg can build the change?

Commit:
xfce.xfce4-power-manager: use fixed path for helpers

Using a nix store path for `xfpm-power-backlight-helper` and `xfce4-pm-helper` causes permission
errors after a rebuild when adjusting brightness or putting the system to sleep, because the
permitted executables are different from the ones `xfce4-power-manager` wants to execute.  Using a
fixed path (/run/current-system/sw/bin) means the paths do not change after a rebuild, so there's no
permission error.

A similar approach is taken with gnome's `gsd-backlight-helper`.

Fixes #77485

pjstadig commented:
@bobby285271 I made the changes
